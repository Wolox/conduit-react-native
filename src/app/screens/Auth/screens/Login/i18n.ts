import i18next from 'i18next';

i18next.addResources('en', 'LOGIN', {
  MAIL: 'Email',
  MAIL_PLACEHOLDER: 'Ej: email@dominio.com',
  PASSWORD: 'Password',
  LOG_IN: 'Sign In',
  LOGIN_FAILURE: 'Email or passsword invalid',
  SIGN_UP: 'Â¿Need an account? Register!'
});
