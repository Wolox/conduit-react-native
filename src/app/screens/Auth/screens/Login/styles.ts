import { StyleSheet } from 'react-native';
import { green, lightGray, white } from '@constants/colors';

const COMMON_VALUE_25 = 25;
export default StyleSheet.create({
  container: {
    alignItems: 'center',
    flex: 1,
    justifyContent: 'center',
    width: '100%'
  },
  form: {
    width: 250
  },
  input: {
    borderBottomWidth: 0,
    backgroundColor: white,
    width: 300,
    height: 50,
    borderRadius: 15,
    elevation: 1,
    shadowColor: lightGray,
    shadowOpacity: 1,
    shadowOffset: {
      width: 0,
      height: 3
    },
    alignSelf: 'center',
    paddingHorizontal: 8
  },
  errorContainer: {
    width: 280,
    alignSelf: 'center',
    textAlign: 'left'
  },
  inputText: {
    padding: 10
  },
  labelText: {
    alignSelf: 'center',
    color: green,
    width: 290,
    textAlign: 'left'
  },
  formButton: {
    backgroundColor: green,
    padding: 10,
    borderRadius: COMMON_VALUE_25,
    marginTop: 2
  },
  textFormButton: {
    color: white,
    fontWeight: 'bold',
    width: '60%',
    textAlign: 'center'
  },
  textFormCustomText: {
    marginTop: COMMON_VALUE_25
  }
});
