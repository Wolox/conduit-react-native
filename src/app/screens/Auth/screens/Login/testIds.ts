export default {
  loginView: 'login-view',
  loginButton: 'login-button',
  signUpButton: 'signup-button',
  emailLoginInput: 'email-login-input',
  passwordLoginInput: 'password-login-input'
};
