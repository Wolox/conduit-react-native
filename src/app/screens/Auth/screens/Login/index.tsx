import React from 'react';
import { Keyboard, TouchableWithoutFeedback, View } from 'react-native';
import { useSelector, useDispatch } from 'react-redux';
import i18next from 'i18next';
import { useForm } from 'react-hook-form';
import CustomButton from '@components/CustomButton';
import CustomText from '@components/CustomText';
import ControlledCustomTextInput from '@components/CustomTextInput/controller';
import Routes from '@constants/routes';
import { Navigation } from '@interfaces/navigation';
import { State } from '@interfaces/reduxInterfaces';
import { actionCreators as AuthActions } from '@redux/auth/actions';
import { FIELDS, LoginFormValues } from '@screens/Auth/constants';
import { validateRequired, validateEmail } from '@utils/validations/validateUtils';

import './i18n';
import styles from './styles';
import testIds from './testIds';

function Login({ navigation }: Navigation) {
  const dispatch = useDispatch();
  const hasLoginError = useSelector<State, boolean>((state: State) => !!state.auth.currentUserError);

  const { handleSubmit, control } = useForm<LoginFormValues>({ mode: 'onBlur' });

  const handleLogin = (values: LoginFormValues) => dispatch(AuthActions.login(values));

  const handleGoToSignUp = () => navigation.navigate(Routes.SignUp);
  return (
    <TouchableWithoutFeedback onPress={Keyboard.dismiss} testID={testIds.loginView}>
      <View style={styles.container}>
        <View style={styles.form}>
          <ControlledCustomTextInput
            testIDProp={testIds.emailLoginInput}
            control={control}
            animated
            keyboardType="email-address"
            label={i18next.t('LOGIN:MAIL')}
            name={FIELDS.email}
            placeholder={i18next.t('LOGIN:MAIL_PLACEHOLDER')}
            showError={hasLoginError}
            rules={{ ...validateRequired, ...validateEmail }}
          />
          <ControlledCustomTextInput
            testIDProp={testIds.passwordLoginInput}
            control={control}
            animated
            showEye
            secureTextEntry
            label={i18next.t('LOGIN:PASSWORD')}
            name={FIELDS.password}
            showError={hasLoginError}
            rules={validateRequired}
          />
          {hasLoginError && (
            <CustomText error center>
              {i18next.t('LOGIN:LOGIN_FAILURE')}
            </CustomText>
          )}
        </View>
        <CustomButton
          testID={testIds.loginButton}
          onPress={handleSubmit(handleLogin)}
          style={styles.formButton}
          title={i18next.t('LOGIN:LOG_IN')}
        />
        <CustomButton
          testID={testIds.signUpButton}
          onPress={handleGoToSignUp}
          style={styles.formButton}
          title={i18next.t('LOGIN:SIGN_UP')}
        />
      </View>
    </TouchableWithoutFeedback>
  );
}

export default Login;
