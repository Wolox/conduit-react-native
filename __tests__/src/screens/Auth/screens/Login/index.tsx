import React from 'react';
import renderer from 'react-test-renderer';
import MockedNavigator from '@mocks/MockedNavigator';
import Login from '@screens/Auth/screens/Login/index';

jest.mock('react-redux', () => {
  return {
    useSelector: jest.fn().mockImplementation(() => ({})),
    useDispatch: () => jest.fn()
  };
});

describe('test LOGING_SCREEN', () => {
  test('to match with snapshot', () => {
    const tree = renderer.create(<MockedNavigator component={Login} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
