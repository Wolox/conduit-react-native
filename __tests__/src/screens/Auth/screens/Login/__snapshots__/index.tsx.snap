// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test LOGING_SCREEN to match with snapshot 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <RNCSafeAreaProvider
    onInsetsChange={[Function]}
    style={
      Array [
        Object {
          "flex": 1,
        },
        undefined,
      ]
    }
  >
    <View
      pointerEvents="box-none"
      style={
        Object {
          "zIndex": 1,
        }
      }
    >
      <View
        accessibilityElementsHidden={false}
        importantForAccessibility="auto"
        onLayout={[Function]}
        pointerEvents="box-none"
        style={null}
      >
        <View
          pointerEvents="box-none"
          style={
            Object {
              "bottom": 0,
              "left": 0,
              "opacity": 1,
              "position": "absolute",
              "right": 0,
              "top": 0,
              "zIndex": 0,
            }
          }
        >
          <View
            style={
              Object {
                "backgroundColor": "rgb(255, 255, 255)",
                "borderBottomColor": "rgb(216, 216, 216)",
                "flex": 1,
                "shadowColor": "rgb(216, 216, 216)",
                "shadowOffset": Object {
                  "height": 0.5,
                  "width": 0,
                },
                "shadowOpacity": 0.85,
                "shadowRadius": 0,
              }
            }
          />
        </View>
        <View
          pointerEvents="box-none"
          style={
            Object {
              "height": 64,
              "maxHeight": undefined,
              "minHeight": undefined,
              "opacity": undefined,
              "transform": undefined,
            }
          }
        >
          <View
            pointerEvents="none"
            style={
              Object {
                "height": 20,
              }
            }
          />
          <View
            pointerEvents="box-none"
            style={
              Object {
                "alignItems": "center",
                "flex": 1,
                "flexDirection": "row",
                "justifyContent": "center",
              }
            }
          >
            <View
              pointerEvents="box-none"
              style={
                Object {
                  "marginHorizontal": 16,
                  "opacity": 1,
                }
              }
            >
              <Text
                accessibilityRole="header"
                aria-level="1"
                numberOfLines={1}
                onLayout={[Function]}
                style={
                  Object {
                    "color": "rgb(28, 28, 30)",
                    "fontSize": 17,
                    "fontWeight": "600",
                  }
                }
              >
                MockedScreen
              </Text>
            </View>
          </View>
        </View>
      </View>
    </View>
    <RNSScreenContainer
      onLayout={[Function]}
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <RNSScreen
        activityState={2}
        forwardedRef={[Function]}
        pointerEvents="box-none"
        style={
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
      >
        <View
          collapsable={false}
          style={
            Object {
              "opacity": 1,
            }
          }
        />
        <View
          accessibilityElementsHidden={false}
          closing={false}
          gestureVelocityImpact={0.3}
          importantForAccessibility="auto"
          onClose={[Function]}
          onGestureBegin={[Function]}
          onGestureCanceled={[Function]}
          onGestureEnd={[Function]}
          onOpen={[Function]}
          onTransition={[Function]}
          pointerEvents="box-none"
          style={
            Array [
              Object {
                "overflow": undefined,
              },
              Object {
                "bottom": 0,
                "left": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
              },
            ]
          }
          transitionSpec={
            Object {
              "close": Object {
                "animation": "spring",
                "config": Object {
                  "damping": 500,
                  "mass": 3,
                  "overshootClamping": true,
                  "restDisplacementThreshold": 10,
                  "restSpeedThreshold": 10,
                  "stiffness": 1000,
                },
              },
              "open": Object {
                "animation": "spring",
                "config": Object {
                  "damping": 500,
                  "mass": 3,
                  "overshootClamping": true,
                  "restDisplacementThreshold": 10,
                  "restSpeedThreshold": 10,
                  "stiffness": 1000,
                },
              },
            }
          }
        >
          <View
            needsOffscreenAlphaCompositing={false}
            pointerEvents="box-none"
            style={
              Object {
                "flex": 1,
              }
            }
          >
            <View
              collapsable={false}
              forwardedRef={[Function]}
              onGestureHandlerEvent={[Function]}
              onGestureHandlerStateChange={[Function]}
              style={
                Object {
                  "flex": 1,
                  "transform": Array [
                    Object {
                      "translateX": 0,
                    },
                    Object {
                      "translateX": 0,
                    },
                  ],
                }
              }
            >
              <View
                pointerEvents="none"
                style={
                  Object {
                    "backgroundColor": "rgb(242, 242, 242)",
                    "bottom": 0,
                    "left": 0,
                    "position": "absolute",
                    "shadowColor": "#000",
                    "shadowOffset": Object {
                      "height": 1,
                      "width": -1,
                    },
                    "shadowOpacity": 0.3,
                    "shadowRadius": 5,
                    "top": 0,
                    "width": 3,
                  }
                }
              />
              <View
                style={
                  Array [
                    Object {
                      "flex": 1,
                      "overflow": "hidden",
                    },
                    Array [
                      Object {
                        "backgroundColor": "rgb(242, 242, 242)",
                      },
                      undefined,
                    ],
                  ]
                }
              >
                <View
                  style={
                    Object {
                      "flex": 1,
                      "flexDirection": "column-reverse",
                    }
                  }
                >
                  <View
                    style={
                      Object {
                        "flex": 1,
                      }
                    }
                  >
                    <View
                      accessible={true}
                      focusable={true}
                      onClick={[Function]}
                      onResponderGrant={[Function]}
                      onResponderMove={[Function]}
                      onResponderRelease={[Function]}
                      onResponderTerminate={[Function]}
                      onResponderTerminationRequest={[Function]}
                      onStartShouldSetResponder={[Function]}
                      style={
                        Object {
                          "alignItems": "center",
                          "flex": 1,
                          "justifyContent": "center",
                          "width": "100%",
                        }
                      }
                      testID="login-view"
                    >
                      <Image
                        source={
                          Object {
                            "process": [Function],
                          }
                        }
                        style={
                          Object {
                            "height": 200,
                            "position": "absolute",
                            "top": 0,
                            "width": "100%",
                          }
                        }
                      />
                      <View
                        style={
                          Object {
                            "alignItems": "center",
                            "height": 50,
                            "justifyContent": "center",
                            "marginBottom": 5,
                            "marginTop": 30,
                            "width": "50%",
                          }
                        }
                      >
                        <Image
                          source={
                            Object {
                              "process": [Function],
                            }
                          }
                          style={
                            Object {
                              "height": "60%",
                              "position": "absolute",
                              "top": 0,
                              "width": 140,
                            }
                          }
                        />
                      </View>
                      <View
                        style={
                          Object {
                            "width": 250,
                          }
                        }
                      >
                        <View
                          style={
                            Array [
                              Object {
                                "marginBottom": 5,
                                "padding": 10,
                              },
                              false,
                              Object {
                                "padding": 10,
                              },
                            ]
                          }
                        >
                          <Text
                            style={
                              Array [
                                Object {
                                  "backgroundColor": "transparent",
                                  "color": "#000",
                                  "fontFamily": "Nunito",
                                  "fontSize": 25.142857142857142,
                                  "fontStyle": "normal",
                                  "fontWeight": "400",
                                },
                                Array [
                                  Object {
                                    "color": "#A7A7A7",
                                  },
                                  Object {
                                    "fontSize": 22,
                                  },
                                ],
                                Array [
                                  Object {
                                    "backgroundColor": "transparent",
                                    "color": "#000",
                                    "fontFamily": "Nunito",
                                    "fontSize": 12,
                                    "fontStyle": "normal",
                                    "fontWeight": "400",
                                    "marginTop": 5,
                                  },
                                  Object {
                                    "alignSelf": "center",
                                    "color": "#5CB85C",
                                    "textAlign": "left",
                                    "width": 290,
                                  },
                                ],
                              ]
                            }
                          >
                            LOGIN:MAIL
                          </Text>
                          <View
                            style={
                              Array [
                                Object {
                                  "alignItems": "center",
                                  "alignSelf": "center",
                                  "backgroundColor": "#FFF",
                                  "borderBottomWidth": 0,
                                  "borderColor": "#555555",
                                  "borderRadius": 15,
                                  "elevation": 1,
                                  "flexDirection": "row",
                                  "height": 50,
                                  "justifyContent": "space-between",
                                  "paddingHorizontal": 8,
                                  "shadowColor": "#E5E5E5",
                                  "shadowOffset": Object {
                                    "height": 3,
                                    "width": 0,
                                  },
                                  "shadowOpacity": 1,
                                },
                                Object {
                                  "borderColor": "#FF0000",
                                },
                                Object {
                                  "alignSelf": "center",
                                  "backgroundColor": "#FFF",
                                  "borderBottomWidth": 0,
                                  "borderRadius": 15,
                                  "elevation": 1,
                                  "height": 50,
                                  "paddingHorizontal": 8,
                                  "shadowColor": "#E5E5E5",
                                  "shadowOffset": Object {
                                    "height": 3,
                                    "width": 0,
                                  },
                                  "shadowOpacity": 1,
                                  "width": 300,
                                },
                                Object {
                                  "alignSelf": "center",
                                  "backgroundColor": "#FFF",
                                  "borderBottomWidth": 0,
                                  "borderRadius": 15,
                                  "elevation": 1,
                                  "height": 50,
                                  "paddingHorizontal": 8,
                                  "shadowColor": "#E5E5E5",
                                  "shadowOffset": Object {
                                    "height": 3,
                                    "width": 0,
                                  },
                                  "shadowOpacity": 1,
                                  "width": 300,
                                },
                              ]
                            }
                          >
                            <TextInput
                              allowFontScaling={false}
                              autoCapitalize="none"
                              autoCompleteType="off"
                              autoCorrect={false}
                              clearButtonMode="never"
                              editable={true}
                              keyboardType="email-address"
                              multiline={false}
                              name="email"
                              onBlur={[Function]}
                              onChangeText={[Function]}
                              onFocus={[Function]}
                              placeholder="LOGIN:MAIL_PLACEHOLDER"
                              rejectResponderTermination={true}
                              returnKeyType="done"
                              style={
                                Array [
                                  Object {
                                    "color": "#555555",
                                    "fontFamily": "Nunito",
                                    "fontSize": 14,
                                    "fontStyle": "normal",
                                    "fontWeight": "400",
                                    "margin": 0,
                                    "padding": 0,
                                  },
                                  Object {
                                    "flex": 1,
                                  },
                                  undefined,
                                  undefined,
                                ]
                              }
                              testID="email-login-input"
                              underlineColorAndroid="transparent"
                              value=""
                            />
                          </View>
                          <View
                            style={
                              Array [
                                Object {
                                  "height": 15,
                                  "marginTop": 3,
                                },
                                Object {
                                  "alignSelf": "center",
                                  "textAlign": "left",
                                  "width": 280,
                                },
                              ]
                            }
                          />
                        </View>
                        <View
                          style={
                            Array [
                              Object {
                                "marginBottom": 5,
                                "padding": 10,
                              },
                              false,
                              Object {
                                "padding": 10,
                              },
                            ]
                          }
                        >
                          <Text
                            style={
                              Array [
                                Object {
                                  "backgroundColor": "transparent",
                                  "color": "#000",
                                  "fontFamily": "Nunito",
                                  "fontSize": 25.142857142857142,
                                  "fontStyle": "normal",
                                  "fontWeight": "400",
                                },
                                Array [
                                  Object {
                                    "color": "#A7A7A7",
                                  },
                                  Object {
                                    "fontSize": 22,
                                  },
                                ],
                                Array [
                                  Object {
                                    "backgroundColor": "transparent",
                                    "color": "#000",
                                    "fontFamily": "Nunito",
                                    "fontSize": 12,
                                    "fontStyle": "normal",
                                    "fontWeight": "400",
                                    "marginTop": 5,
                                  },
                                  Object {
                                    "alignSelf": "center",
                                    "color": "#5CB85C",
                                    "textAlign": "left",
                                    "width": 290,
                                  },
                                ],
                              ]
                            }
                          >
                            LOGIN:PASSWORD
                          </Text>
                          <View
                            style={
                              Array [
                                Object {
                                  "alignItems": "center",
                                  "alignSelf": "center",
                                  "backgroundColor": "#FFF",
                                  "borderBottomWidth": 0,
                                  "borderColor": "#555555",
                                  "borderRadius": 15,
                                  "elevation": 1,
                                  "flexDirection": "row",
                                  "height": 50,
                                  "justifyContent": "space-between",
                                  "paddingHorizontal": 8,
                                  "shadowColor": "#E5E5E5",
                                  "shadowOffset": Object {
                                    "height": 3,
                                    "width": 0,
                                  },
                                  "shadowOpacity": 1,
                                },
                                Object {
                                  "borderColor": "#FF0000",
                                },
                                Object {
                                  "alignSelf": "center",
                                  "backgroundColor": "#FFF",
                                  "borderBottomWidth": 0,
                                  "borderRadius": 15,
                                  "elevation": 1,
                                  "height": 50,
                                  "paddingHorizontal": 8,
                                  "shadowColor": "#E5E5E5",
                                  "shadowOffset": Object {
                                    "height": 3,
                                    "width": 0,
                                  },
                                  "shadowOpacity": 1,
                                  "width": 300,
                                },
                                Object {
                                  "alignSelf": "center",
                                  "backgroundColor": "#FFF",
                                  "borderBottomWidth": 0,
                                  "borderRadius": 15,
                                  "elevation": 1,
                                  "height": 50,
                                  "paddingHorizontal": 8,
                                  "shadowColor": "#E5E5E5",
                                  "shadowOffset": Object {
                                    "height": 3,
                                    "width": 0,
                                  },
                                  "shadowOpacity": 1,
                                  "width": 300,
                                },
                              ]
                            }
                          >
                            <TextInput
                              allowFontScaling={false}
                              autoCapitalize="none"
                              autoCompleteType="off"
                              autoCorrect={false}
                              clearButtonMode="never"
                              editable={true}
                              keyboardType="default"
                              multiline={false}
                              name="password"
                              onBlur={[Function]}
                              onChangeText={[Function]}
                              onFocus={[Function]}
                              placeholder="LOGIN:PASSWORD"
                              rejectResponderTermination={true}
                              returnKeyType="done"
                              secureTextEntry={true}
                              style={
                                Array [
                                  Object {
                                    "color": "#555555",
                                    "fontFamily": "Nunito",
                                    "fontSize": 14,
                                    "fontStyle": "normal",
                                    "fontWeight": "400",
                                    "margin": 0,
                                    "padding": 0,
                                  },
                                  Object {
                                    "flex": 1,
                                  },
                                  undefined,
                                  undefined,
                                ]
                              }
                              testID="password-login-input"
                              underlineColorAndroid="transparent"
                              value=""
                            />
                            <View
                              accessible={true}
                              focusable={true}
                              onClick={[Function]}
                              onResponderGrant={[Function]}
                              onResponderMove={[Function]}
                              onResponderRelease={[Function]}
                              onResponderTerminate={[Function]}
                              onResponderTerminationRequest={[Function]}
                              onStartShouldSetResponder={[Function]}
                              style={
                                Object {
                                  "alignItems": "center",
                                  "display": "flex",
                                  "flexDirection": "row",
                                  "opacity": 1,
                                }
                              }
                            >
                              <Image
                                resizeMode="contain"
                                source={
                                  Object {
                                    "process": [Function],
                                  }
                                }
                                style={
                                  Object {
                                    "marginLeft": 8,
                                    "width": 20,
                                  }
                                }
                              />
                            </View>
                          </View>
                          <View
                            style={
                              Array [
                                Object {
                                  "height": 15,
                                  "marginTop": 3,
                                },
                                Object {
                                  "alignSelf": "center",
                                  "textAlign": "left",
                                  "width": 280,
                                },
                              ]
                            }
                          />
                        </View>
                        <Text
                          style={
                            Array [
                              Object {
                                "backgroundColor": "transparent",
                                "color": "#000",
                                "fontFamily": "Nunito",
                                "fontSize": 25.142857142857142,
                                "fontStyle": "normal",
                                "fontWeight": "400",
                              },
                              Array [
                                Object {
                                  "textAlign": "center",
                                },
                                Object {
                                  "color": "#FF0000",
                                },
                              ],
                              undefined,
                            ]
                          }
                        >
                          LOGIN:LOGIN_FAILURE
                        </Text>
                      </View>
                      <View
                        accessible={true}
                        focusable={true}
                        onClick={[Function]}
                        onResponderGrant={[Function]}
                        onResponderMove={[Function]}
                        onResponderRelease={[Function]}
                        onResponderTerminate={[Function]}
                        onResponderTerminationRequest={[Function]}
                        onStartShouldSetResponder={[Function]}
                        style={
                          Object {
                            "alignItems": "center",
                            "backgroundColor": "#5CB85C",
                            "borderColor": "#FFF",
                            "borderRadius": 25,
                            "borderWidth": 0,
                            "flexDirection": "row",
                            "justifyContent": "center",
                            "marginTop": 2,
                            "opacity": 1,
                            "padding": 10,
                          }
                        }
                        testID="login-button"
                      >
                        <Text
                          style={
                            Array [
                              Object {
                                "backgroundColor": "transparent",
                                "color": "#000",
                                "fontFamily": "Nunito",
                                "fontSize": 25.142857142857142,
                                "fontStyle": "normal",
                                "fontWeight": "400",
                              },
                              Array [],
                              Array [
                                Array [],
                                Object {
                                  "color": "#FFF",
                                  "fontWeight": "bold",
                                  "textAlign": "center",
                                  "width": "60%",
                                },
                                Object {
                                  "color": "#FFF",
                                },
                              ],
                            ]
                          }
                        >
                          LOGIN:LOG_IN
                        </Text>
                      </View>
                      <View
                        accessible={true}
                        focusable={true}
                        onClick={[Function]}
                        onResponderGrant={[Function]}
                        onResponderMove={[Function]}
                        onResponderRelease={[Function]}
                        onResponderTerminate={[Function]}
                        onResponderTerminationRequest={[Function]}
                        onStartShouldSetResponder={[Function]}
                        style={
                          Object {
                            "marginTop": 25,
                            "opacity": 1,
                          }
                        }
                      >
                        <Text
                          style={
                            Array [
                              Object {
                                "backgroundColor": "transparent",
                                "color": "#000",
                                "fontFamily": "Nunito",
                                "fontSize": 25.142857142857142,
                                "fontStyle": "normal",
                                "fontWeight": "400",
                              },
                              Array [
                                Object {
                                  "color": "#5CB85C",
                                },
                                Object {
                                  "fontSize": 18.857142857142858,
                                },
                              ],
                              undefined,
                            ]
                          }
                        >
                          LOGIN:SIGN_UP
                        </Text>
                      </View>
                    </View>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </View>
      </RNSScreen>
    </RNSScreenContainer>
  </RNCSafeAreaProvider>
</View>
`;
